src/main/java/com/hierynomus/msfscc/fileinformation/FileInformationFactory.java

```diff
@@ -37,7 +38,7 @@
         renBuf.putRawBytes(new byte[]{0, 0, 0, 0, 0, 0, 0});
         renBuf.putUInt64(0);
         renBuf.putUInt32(newName.length() * 2); // unicode
-        renBuf.putRawBytes(newName.getBytes(StandardCharsets.UTF_16));
+        renBuf.putRawBytes(SMB2Functions.unicode(newName));
```

Comment:
Ok, I will change it to `renBuf.putRawBytes(newName.getBytes(StandardCharsets.UTF_16LE)`.